networkit/cpp/linkprediction/AdjustedRandIndex.cpp

```diff
@@ -6,15 +6,14 @@
  *      Author: Kolja Esders
  */
 
-#include <set>
+#include <unordered_set>
 
 #include <networkit/linkprediction/AdjustedRandIndex.hpp>
 
 namespace NetworKit {
 
 double AdjustedRandIndex::runImpl(node u, node v) {
-  std::set<node> uNeighbors;
-  std::set<node> vNeighbors;
+  std::unordered_set<node> uNeighbors, vNeighbors;
```

Comment:
Running `set_intersection` (which needs a sorted range) on an unordered set looks wrong. I think this is introducing a bug. Using `std::sort` on a `std::vector` might be more efficient, though. Reading the code further, I think there should be a set difference instead of a union, though after knowing the size of the intersection, it should be possible to calculate all of these numbers without further set operations.